common/vts.proto

```diff
@@ -11,109 +11,52 @@ import "attestation.proto";
 import "token.proto";
 
 
-enum TAType {
-    TA_UNSPECIFIED = 0;
-    TA_RAWPUBLICKEY = 1;
-    // Refer https://github.com/veraison/veraison/issues/76, to introduce certs here!
-    TA_INVALID = 255;
-}
-
-message EndorsementID {
-    AttestationFormat type = 1;
-    google.protobuf.Struct parts = 2;
-}
-
-message Evidence {
-    google.protobuf.Struct value = 1;
-}
-
-message GetEndorsementsRequest{
-    EndorsementID id = 1;
-    Evidence evidence = 2;
-}
-
 message Status {
     bool result = 1;
     string error_detail = 2;
 }
 
-message GetEndorsementsResponse {
-    Status status = 1;
-    string details = 2;
-    TrustVector trust_vector = 3;
-    google.protobuf.Struct endorsements = 4;
-}
-
-message SwComponentID {
-    AttestationFormat type = 1;
-    google.protobuf.Struct parts = 2;
-}
-
-message SwComponent {
-    SwComponentID id = 1;
-    google.protobuf.Struct attributes = 2;
+message Evidence {
+    google.protobuf.Struct value = 1;
 }
 
 message AddSwComponentsRequest {
-    repeated SwComponent info = 1;
+    repeated Endorsement sw_components = 1;
 }
 
 message AddSwComponentsResponse {
     Status status = 1;
 }
 
-message GetSwComponentRequest {
-    SwComponentID id = 1;
-}
-
-message GetSwComponentResponse {
-    Status status = 1;
-    google.protobuf.Struct attributes = 2;
-}
-
-message TrustAnchorID {
-    AttestationFormat type = 1;
-    google.protobuf.Struct parts = 2;
+enum EndorsementType {
+    UNSET = 0;
+    REFERENCE_VALUE = 1;
+    VERIFICATION_KEY = 2;
 }
 
-message TrustAnchorValue {
-    TAType type = 1;
-    google.protobuf.Struct value = 2;
-}
-
-message TrustAnchor {
-    TrustAnchorID id = 1;
-    TrustAnchorValue value = 2;
+message Endorsement {
+    AttestationFormat scheme = 1;
+    EndorsementType type = 2;
+    google.protobuf.Struct attributes = 3;
 }
 
 message AddTrustAnchorRequest {
-    TrustAnchor trust_anchor = 1;
+    Endorsement trust_anchor = 1;
 }
 
 message AddTrustAnchorResponse {
     Status status = 1;
 }
 
-message GetTrustAnchorRequest {
-    TrustAnchorID ta_id = 1;
-}
-
-message GetTrustAnchorResponse {
-    Status status = 1;
-    TrustAnchorValue value = 2;
-}
-
 // Client interface for the Veraison Trusted Services component.
 service VTS {
     // Returns attestation information -- evidences, endorsed claims, trust
     // vector, etc -- for the provided attestation token data.
     rpc GetAttestation(AttestationToken) returns (Attestation);
 
-    // Service endpoints that are used to store and retrieve Software Components
-    // and Trust Anchors to and from the endorsement store
+    // Service endpoints that are used to store Software Components
+    // and Trust Anchors to the endorsement store
```

Comment:
Thinking a bit more, as this is a generic interface and not tied to any specific attestation scheme:
Ideally this should not be named as SW Components, but AddReferenceVals()

Reason been, in a particular attestation scheme, the attestation may be other types of reference values which we want to provision but they may not be SW Components!

Reply:
Yeah, that may be a slightly better name.

Reply:
Anyway, I suggest we have a separate "naming consistency" conversation (where we may also want to discuss things like using "Evidence" instead of "Attestation Token" in the verification pipeline) and unblock this PR.

Reply:
how big is an impact just of changing SwComponent in the interface to RefVal ? I agree for other bits, we need a different PR, but can we consider, the impact before concluding this change also (via separate PR?)

Reply:
The impact is a search and replace on ~20 files.

Reply:
> The distinction is important in the sense that they are supposed to go into different store buckets. The plugin knows where they need to be routed.

This can be done based on the "type" field of the Endorsement,  and does not necessarily require a separate method. Using a single message would actually be more flexible, as there this means we don't actually have to define the number of stores architecturally -- this can be left to the implementation (e.g. what is the issue with keeping trust anchors and ensorsements/refvals in the same store?)

Reply:
For addressing this specific concern I have raised Issue #151. However , in the vts.proto SwComponent to RefVal change is quite a logical change which can go first with this change, unless you feel need a more careful deliberation?

Reply:
Please refer Issue #152 for further Architectural Discussion